### ISSUE TITLE
probe: how to handle older measurements inside the in-app DB

### ISSUE THREAD

**Issue by hellais** [ooni/probe#1983]
It has been reported in https://github.com/ooni/probe/issues/1950 & https://github.com/ooni/probe/issues/1927 that it's a problem for users that run OONI Probe unattended for a long time that the database size just keeps growing.

Currently whenever a measurement is uploaded we delete the raw JSON on disk and store in the local database the metadata about the measurement so that we can display it in the result listing. If the measurement is not uploaded, in addition to the metadata in the DB, we also retain the raw JSON.

What is being proposed here is that, in addition to deleting the raw JSON once it’s uploaded (which is already done), we also delete the metadata from the database after some amount of time. If we do this, those older measurements will effectively disappear from the users app and will not be retrievable in any other way anymore.

We should have an option in the settings that will automatically delete measurements from the DB that are older than some predefined date interval.

This is also related to: https://github.com/ooni/probe/issues/1031

**Comment by hellais** [comment]
We are probably going to do something similar now that autorun is shipped in multiplatform, though since we fetch them remotely probably we don't have to delete everything.


Adding some notes about this from slack:

For doing something automatic, that's why I suggested clearing automatically results older than 1 year. We can let let users know in the onboarding on in the settings. And if we get complains, we can turn it into a Setting in the future.


Maybe this is something worth adding to some user survey we could do in the future. Given that uploaded data is in explorer having older than even 1 month data on your device is probably not so helpful.

With the probe id work we could even have pages in explorer where users could lookup all their measurements there directly and we don’t even have to implement all these UI elements in the app

Maybe one way to look at it could be to say we always delete by default unless:
* measurement hasn’t been uploaded
* the user explicitly marked a measurement as “keep”
For all other cases we provide an in app view to explorer to see all their measurements

It’s anyways going to be more effective to look at a bunch of measurements from your computer with explorer where you can make charts using mat and have more screen real-estate

**Comment by hellais** [comment]
Also useful metrics for multiplatform are:

Each new full OONI run is increasing the size of the database by around 12 KB. It gets slightly smaller as URLs get reused, but it's a negligible reduction.
Assuming auto-run is enabled and every run is successful, that's 288 KB per day, 8.6 MB per month, 105 MB per year.
* Does this size increase warrant any additional action?
* Users right now can only clear everything, there's no selective deletion.
Should we automatically delete data older than 1 year? And/Or add that as a setting?
I don't think it's something we need to address right away. But since we're now storing auto-runs, it's something we should keep in the back of our minds.

**Comment by hellais** [comment]
See also: https://docs.google.com/document/d/1bsDAdA80i4z3Prs-P-88IoNYqb7LIL9TkYlgLQgDG0M/edit?tab=t.0#heading=h.y0bgdagcsvoc

**Comment by sdsantos** [comment]
@hellais Following-up on this. No complains from users so far about the new release. Should we move forward with some of the other mitigations proposed?

a) Allow users to selectively clear results.
b) Automatically clear results older than 6 months or 1 year (or make it a settings option).
c) Add a filter by start and end date, so users can always find a certain result in the database.

At least b) sounds like an easy thing to implement that could mitigate potential issues for users in a couple of months.

**Comment by hellais** [comment]
Yeah let's start off with b), but apply this auto-cleanup only to results that have not already been uploaded. If they haven't been uploaded we should not auto-delete it.

We can expose this as a setting with a default of 6 months.

**Comment by sdsantos** [comment]
@hellais `but apply this auto-cleanup only to results that have not already been uploaded` you mean that have been uploaded right? That's what you say afterwards...

Also, what kind of setting should it be? The amount of months to keep, with a minimum of 1 month?
